src/win.c

```diff
@@ -461,6 +478,203 @@ static void win_update_properties(session_t *ps, struct managed_win *w) {
 	win_clear_all_properties_stale(w);
 }
 
+static void init_animation(session_t *ps, struct managed_win *w) {
+	CLEAR_MASK(w->in_desktop_animation)
+	static int32_t randr_mon_center_x, randr_mon_center_y;
+	if (w->randr_monitor == -1) {
+		win_update_monitor(ps->randr_nmonitors, ps->randr_monitor_regs, w);
+		if (w->randr_monitor != -1) {
+			auto e = pixman_region32_extents(
+			    &ps->randr_monitor_regs[w->randr_monitor]);
+			randr_mon_center_x = (e->x2 + e->x1) / 2, randr_mon_center_y =
+			                                              (e->y2 + e->y1) / 2;
+		} else {
+			randr_mon_center_x = ps->root_width / 2, randr_mon_center_y =
+			                                             ps->root_height / 2;
+		}
+	} else {
+		auto e = pixman_region32_extents(&ps->randr_monitor_regs[w->randr_monitor]);
+		randr_mon_center_x = (e->x2 + e->x1) / 2, randr_mon_center_y =
+		                                              (e->y2 + e->y1) / 2;
+	}
+	static double *anim_x, *anim_y, *anim_w, *anim_h;
+	enum open_window_animation animation;
+	bool transient_window = is_transient(ps, w);
+
+	if (w->in_desktop_animation || w->window_type == WINTYPE_DESKTOP) {
+		animation = OPEN_WINDOW_ANIMATION_NONE;
+	} else {
+		animation = ps->o.wintype_option[w->window_type].animation;
+	}
+
+	/* START OF ANIMATION CODE (make it optional) */
+
+	if (animation == OPEN_WINDOW_ANIMATION_INVALID) {
+		if (transient_window) {
+			int16_t mx, my;
+			animation = OPEN_WINDOW_ANIMATION_SLIDE_DOWN; //ps->o.animation_for_transient_window;
+			if (get_mouse_position(ps, &mx, &my)) {
+				if (my >= (randr_mon_center_y*2) - w->pending_g.height) {
+					animation = OPEN_WINDOW_ANIMATION_SLIDE_UP;
```

Comment:
here too

Reply:
Yes, same purpose... I really like those animations but maybe it breaks on some toolkits / WM so I left the boring way in comments.